src/btree/bt_slvg.c

```diff
@@ -596,13 +603,17 @@ __slvg_trk_leaf(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, uint8_t *ad
           trk->col_stop);
         break;
     case WT_PAGE_COL_VAR:
+        WT_TIME_AGGREGATE_INIT_MAX(&trk->trk_ta);
+
         /*
          * Column-store variable-length format: the start key can be taken from the block's header,
          * stop key requires walking the page.
          */
         stop_recno = dsk->recno;
         WT_CELL_FOREACH_KV (session, dsk, unpack) {
             stop_recno += __wt_cell_rle(&unpack);
+
+            WT_TIME_AGGREGATE_UPDATE(&trk->trk_ta, &unpack.tw);
```

Comment:
Should we aggregate the tws here? If we rewrite a page, we will do a reconciliation and the time windows should be automatically aggregated there I think.

Reply:
I have a ditched pr https://github.com/wiredtiger/wiredtiger/pull/5703 which relies on reconciliation to aggregate the time windows.

Reply:
This isn't rewriting the page, this is applying the leaf-page record to the aggregate time window for the page.

Reply:
oh, I see.

Reply:
I think this is a good point, @quchenhao.

I added calls to `WT_TIME_AGGREGATE_MERGE` in the two routines that build internal pages (`__slvg_row_build_internal` and `__slvg_col_build_internal`), and I don't think they're necessary. The reconciliation code should do all of that work for me.

The same is true for the `WT_CKPT.write_gen` field, updated with the timestamp information in `__wt_checkpoint_tree_reconcile_update()`.

I've pushed that change in ee8dbd5a7.